File: lab-3-ddb/environment/ruby_2.6.0/add_index.rb
require "aws-sdk-dynamodb"
client = <FMI>(region: "us-east-1")

res = client.update_table(
    attribute_definitions:[
        {
                attribute_name: "breed",
                attribute_type: "S"
        },
    ], 
    table_name: "lostcats",
    <FMI>:[
        {
            create: {
                index_name: 'breed_index',
                <FMI>: [
                    {
                        attribute_name: 'breed',
                        key_type: 'HASH'
                    }
                ],
                projection: {
                    <FMI>: 'ALL'
                },
                provisioned_throughput: {
                    read_capacity_units: 1,
                    write_capacity_units: 1
                }
            }
        }
    ],
)

File: lab-3-ddb/environment/node_8.10.0/add_index.js
var 
    AWS = require("aws-sdk"),                              
    DDB = new AWS.DynamoDB({
        apiVersion: "2012-08-10",
        region: "<FMI>"
    });                                                    

(function addIndex(){
    var 
        params = {
            TableName: "lostcats",
            AttributeDefinitions: [{
                AttributeName: "breed",
                AttributeType: "S"
            }],
            <FMI>: [{
                Create: {
                    <FMI>: "breed_index",
                    KeySchema: [{
                        <FMI>: "breed",
                        KeyType: "<FMI>"
                    }],
                    Projection: {
                        ProjectionType: "<FMI>"
                    },
                    ProvisionedThroughput: {
                        ReadCapacityUnits: 1, 
                        WriteCapacityUnits: 1
                    }
                }
            }]
        };
     DDB.<FMI>(params, function(err, data){
         console.log(err, data);             
     });
})();
